ValidatorError not catched by try/catch

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported


### Mongoose version

^7.5.0

### Node.js version

18.18.0

### MongoDB server version

MongoDB Atlas

### Typescript version (if applicable)

^5.2.2

### Description

The ValidatorError  thrown by mongoose is not catched by try/catch block 
```
try {
    const activity = new Activity({
      <object with invalid schema>
    });

    await activity.save();
    return activity;
  } catch (err) {
    console.log(err);
  }
````

Error:
```
ValidatorError: Path `value` is required.
    at validate (***\node_modules\mongoose\lib\schematype.js:1365:13)
    at SchemaString.SchemaType.doValidate (***\node_modules\mongoose\lib\schematype.js:1349:7)
    at ***\node_modules\mongoose\lib\document.js:2968:18
    at processTicksAndRejections (node:internal/process/task_queues:77:11)
[ERROR] 09:23:29 ValidatorError: Path `value` is required.
```


### Steps to Reproduce

```
try {
    const activity = new Activity({
      <object with invalid schema>
    });

    await activity.save();
    return activity;
  } catch (err) {
    console.log(err);
  }
````

### Expected Behavior

The error should be catched by the closest try/catch block

--- a/lib/schema.js
+++ b/lib/schema.js
@@ -1122,15 +1122,14 @@ Schema.prototype.path = function(path, obj) {
       if (_schemaType.$isMongooseDocumentArray) {
         _schemaType.$embeddedSchemaType._arrayPath = arrayPath;
         _schemaType.$embeddedSchemaType._arrayParentPath = path;
-        _schemaType = _schemaType.$embeddedSchemaType.clone();
+        _schemaType = _schemaType.$embeddedSchemaType;
       } else {
         _schemaType.caster._arrayPath = arrayPath;
         _schemaType.caster._arrayParentPath = path;
-        _schemaType = _schemaType.caster.clone();
+        _schemaType = _schemaType.caster;
       }
 
-      _schemaType.path = arrayPath;
-      toAdd.push(_schemaType);
+      this.subpaths[arrayPath] = _schemaType;
     }
 
     for (const _schemaType of toAdd) {

--- a/test/types.documentarray.test.js
+++ b/test/types.documentarray.test.js
@@ -754,4 +754,28 @@ describe('types.documentarray', function() {
 
     assert.equal(doc.myMap.get('foo').$path(), 'myMap.foo');
   });
+
+  it('bubbles up validation errors from doubly nested doc arrays (gh-14101)', async function() {
+    const optionsSchema = new mongoose.Schema({
+      val: {
+        type: Number,
+        required: true
+      }
+    });
+
+    const testSchema = new mongoose.Schema({
+      name: String,
+      options: {
+        type: [[optionsSchema]],
+        required: true
+      }
+    });
+
+    const Test = db.model('Test', testSchema);
+
+    await assert.rejects(
+      Test.create({ name: 'test', options: [[{ val: null }]] }),
+      /options.0.0.val: Path `val` is required./
+    );
+  });
 });

